### PR TITLE
HOCS-3924: add endpoint for case actions type table query

### PR DESCRIPTION
**Pull request by gary-newbolt**

### CODE REVIEW

File: src/main/java/uk/gov/digital/ho/hocs/info/api/CaseActionsResource.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.digital.ho.hocs.info.api.dto.CaseActionTypeDto;
+
+import java.util.List;
+
+@RestController
+@Slf4j
+public class CaseActionsResource {
```
Comment (agraham19) [inline src/main/java/uk/gov/digital/ho/hocs/info/api/CaseActionsResource.java:15]: We should have @ControllerAdvice with to specify error handling class to handle exceptions gracefully. Have a look at some of the other resource classes.

Comment (agraham19) [reply src/main/java/uk/gov/digital/ho/hocs/info/api/CaseActionsResource.java:15]: Sorry. You don't need an annotation on the resources themselves. My mistake.

File: src/main/java/uk/gov/digital/ho/hocs/info/api/CaseActionsResource.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    private final CaseActionsTypeService caseActionTypeService;
+
+    @Autowired
+    public CaseActionsResource(CaseActionsTypeService caseActionTypeService) {
+        this.caseActionTypeService = caseActionTypeService;
+    }
+
+    @GetMapping("/case-actions/{caseType}")
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/api/CaseActionsResource.java:24]: Wonder if we could have a more 'RESTful' endpoint... `case/actions/{caseType}` or `casetype/{casetype}/actions`.. or similar?

File: src/main/java/uk/gov/digital/ho/hocs/info/api/dto/CaseActionTypeDto.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import uk.gov.digital.ho.hocs.info.domain.model.CaseActionType;
+
+import java.util.UUID;
+
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+@NoArgsConstructor
+@Builder
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/info/api/dto/CaseActionTypeDto.java:14]: Nifty, I didn't know about this...

File: src/main/java/uk/gov/digital/ho/hocs/info/domain/model/CaseActionType.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+@Table(name = "case_action_type")
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class CaseActionType {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
```
[the remark this answers is not in the document]
Comment (LiamMacP) [reply src/main/java/uk/gov/digital/ho/hocs/info/domain/model/CaseActionType.java]: Yep looks good, cheers 🙂

**Pull request merged**